script.js

```diff
@@ -1,4 +1,144 @@
+    }
+
+    var playerNum = retrievePlayerNumber(numberChosen);
+    // console.log(playerNum, "get number");
+    // console.log("p1dice:", diceRollOfP1, "p2dice:", diceRollOfP2);
+    var playerNumResponse =
+      "Player " +
+      currentPlayer +
+      ", You chose Dice " +
+      numberChosen +
+      " first. <br> Your number is " +
+      playerNum +
+      ".";
+
+    if (currentPlayer == 1) {
+      currentPlayer = 2;
+      // console.log(currentPlayer, "current");
+      gameMode = diceGame;
+      // console.log(gameMode, "dice now");
+      return (
+        playerNumResponse +
+        "<br> It is now player 2's turn. Press submit to roll player 2's dice."
+      );
+    }
+    var winingPlayer = getWinner();
+
+    currentPlayer = 1;
+    gameMode = diceGame;
+
+    return (
+      playerNumResponse +
+      "<br> Player " +
+      winingPlayer +
+      " has won. <br> Player 1's number: " +
+      p1Num +
+      " <br> " +
+      " Player 2's number: " +
+      p2Num +
+      " <br><br> Press submit to play again."
+    );
+  }
+};
+
+// dice roll function
+var generateNewDiceRoll = function () {
+  var newDiceRoll = Math.floor(Math.random() * 6) + 1;
+  return newDiceRoll;
+};
+
+// roll dice into array
+var getNewDiceRoll = function () {
+  var newDiceRoll = [generateNewDiceRoll(), generateNewDiceRoll()];
+  // console.log(newDiceRoll, "getnewDiceRoll");
+  if (currentPlayer == 1) {
+    diceRollOfP1 = newDiceRoll;
+    // console.log(diceRollOfP1, "p1roll");
+  } else {
+    diceRollOfP2 = newDiceRoll;
+    // console.log(diceRollOfP2, "p2roll");
+  }
+  return newDiceRoll;
+};
+
+// To get the player's number
+var retrievePlayerNumber = function (numberChosen) {
+  var playerNum = "";
+  // create variable, store the existing dice roll array generated into each player's dice roll array
+  var playerDiceRoll;
+  if (currentPlayer == 1) {
+    playerDiceRoll = diceRollOfP1;
+  } else if (currentPlayer == 2) {
+    playerDiceRoll = diceRollOfP2;
+  }
+
+  if (numberChosen == 1) {
+    // console.log(numberChosen, ", option 1 chosen");
+    if (currentPlayer == 1) {
+      p1Num = "" + playerDiceRoll[0] + playerDiceRoll[1];
+      // console.log(p1Num, "p1num1");
+      return p1Num;
+    } else if (currentPlayer == 2) {
+      p2Num = "" + playerDiceRoll[0] + playerDiceRoll[1];
+      // console.log(p2Num, "p2num2");
+      return p2Num;
+    }
+  } else if (numberChosen == 2) {
+    // console.log(numberChosen, ", option 2 chosen");
+    if (currentPlayer == 1) {
+      p1Num = "" + playerDiceRoll[1] + playerDiceRoll[0];
+      // console.log(p1Num, "p1num3");
+      return p1Num;
+    } else if (currentPlayer == 2) {
+      p2Num = "" + playerDiceRoll[1] + playerDiceRoll[0];
+      // console.log(p2Num, "p2num4");
+      return p2Num;
+    }
+  }
+  return playerNum;
+};
+
+// To figure out who is the winner
+var getWinner = function () {
+  if (p1Num > p2Num) {
+    return 1;
+  } else {
+    return 2;
+  }
```

Comment:
Once again , great use of a helper function. Even though there is very little logic inside, as long as you use a bunch of code multiple times, then it is recommended to create a helper function.